xmake/modules/private/utils/batchcmds.lua

```diff
@@ -231,8 +231,8 @@ function batchcmds:compile(sourcefiles, objectfile, opt)
     local program, argv = compiler_inst:compargv(sourcefiles, objectfile, opt)
 
     -- add compilation command and bind run environments of compiler
-    self:mkdir(path.directory(objectfile))
-    self:vrunv(program, argv, {envs = table.join(compiler_inst:runenvs(), opt.envs)})
+    self:mkdir(path(path.directory(objectfile)))
+    self:vrunv(path(program), argv, {envs = table.join(compiler_inst:runenvs(), opt.envs)})
```

Comment:
program maybe is not an absolute path. e.g. `gcc`, `clang`

Reply:
Even if it is an absolute path, we usually don't need to deal with it because it is not in the project directory